Accept symbol name in extension task

k0kubun opened the pull request:
Since I write rake's normal task name in Symbol, I've used Symbol for the argument of `Rake::ExtensionTask.new` until rake-compiler v0.9.7. But rake-compiler v0.9.8 raises an error with such case in https://github.com/rake-compiler/rake-compiler/blob/v0.9.8/lib/rake/extensiontask.rb#L115.

I propose to accept Symbol for `Rake::ExtensionTask`'s name like `Rake::Task` https://github.com/ruby/rake/blob/v11.1.2/lib/rake/task.rb#L95.

kou commented:
Good catch!
Could you add a test?

kou commented on lib/rake/baseextensiontask.rb:
```diff
@@ -35,7 +35,7 @@ def initialize(name = nil, gem_spec = nil)
 
     def init(name = nil, gem_spec = nil)
-      @name = name
+      @name = name.to_s
```
This change works only for `Task.new(name)` case.
We need to care about `Task#name=` case.

How about stringifying in `#define`?

k0kubun replied:
> This change works only for Task.new(name) case.
> We need to care about Task#name= case.
>
> How about stringifying in #define?

Thank you for nice suggestion. Fixed to stringify name in `#define`. https://github.com/rake-compiler/rake-compiler/pull/134/commits/af7dee631b4011b9f2255c8c83c3ee8ae984c374

k0kubun commented:
> Could you add a test?

Added a test and squashed to https://github.com/rake-compiler/rake-compiler/pull/134/commits/af7dee631b4011b9f2255c8c83c3ee8ae984c374.

kou commented:
Thanks.
I've merged and released a new version.
